### ISSUE TITLE
Proper treatment of cities/geographical locale names

### ISSUE THREAD

**Issue by igorm3** [UniversalDependencies/UD_English-EWT#151]
Hi,

According to the [ud annotation guidelines](https://universaldependencies.org/u/dep/flat.html) `flat` seems to be the most apt relation for handling locales like "Rio de Janeiro", "Sao Paulo", "Sri Lanka", etc., but these consistently get annotated with `compound` relations. Is there a section of the annotation guidelines that is meant to account for this discrepancy or is this a known issue? 

Also, even for English named entities, should "New York" not be tagged with `flat` relations? Thanks in advance for any clarification/guidance on this front.

**Comment by nschneid** [comment]
> According to the [ud annotation guidelines](https://universaldependencies.org/u/dep/flat.html) `flat` seems to be the most apt relation for handling locales like "Rio de Janeiro", "Sao Paulo", "Sri Lanka", etc., but these consistently get annotated with `compound` relations. Is there a section of the annotation guidelines that is meant to account for this discrepancy or is this a known issue?

You are correct, these should be `flat` as they are non-English multiword expressions. This is one of several problems with proper names in EWT.

> Also, even for English named entities, should "New York" not be tagged with `flat` relations? Thanks in advance for any clarification/guidance on this front.

For named entities exhibiting standard English syntax we use relations other than `flat`, in this case `compound`. (You might expect `amod`, but "New" is tagged as a PROPN based on the Penn Treebank tagging guidelines.)

**Comment by amir-zeldes** [comment]
I think these are all `flat` in UD_English-GUM, though you may still find errors here and there. 

As for adjectives tagged PROPN, I think these are generally attached as `amod`, as in:

  * Federal/PROPN/amod Bureau/PROPN/root of/ADP/case Investigation/PROPN/nmod

However specifically for "New York", at least in GUM it is `flat`, since it is considered an opaque flat name expression synchronically (similarly: "New Zealand/flat")

**Comment by nschneid** [comment]
So there's a semantic compositionality criterion for `amod` vs. `compound`? That seems contrary to normal UD practice.

EWT almost always uses compound:

* [EWT PROPN-amod-PROPN relations](http://match.grew.fr/?corpus=UD_English-EWT@dev&custom=607dc7bfd9c8c&clustering=N.xpos): 13 hits
* [GUM PROPN-amod-PROPN relations](http://match.grew.fr/?corpus=UD_English-GUM@dev&custom=607dc77fe8be7&clustering=N.xpos): 446 hits
* [EWT PROPN-compound-PROPN relations](http://match.grew.fr/?corpus=UD_English-EWT@dev&custom=607dc708191cd&clustering=N.xpos): >1000 hits
* [GUM PROPN-compound-PROPN relations](http://match.grew.fr/?corpus=UD_English-GUM@dev&custom=607dc75a55526&clustering=N.xpos): >1000 hits

**Comment by amir-zeldes** [comment]
Ooh, this is actually a big inconsistency! I think the normal UD practice is to analyze syntactic relations "normally", even if they are inside names:

  * "names that have a regular syntactic structure, like The Lord of the Rings and Captured By Aliens, should be annotated with regular syntactic relations."
  * "For organization names with clear syntactic modification structure, the dependencies should also reflect the syntactic modification structure using regular syntactic relations"
https://universaldependencies.org/u/dep/flat.html

The example given in the guidelines is "Natural/PROPN/amod Resources Conservation Service", which I take to be analogous to the FBI example.

The issue of whether "New York" should be amod or flat is a separate one, but I think it is definitely not a compound headed by York...

**Comment by nschneid** [comment]
I think the problem is that we have to decide whether these modifiers are nouns or adjectives. If they're nouns, then compound makes sense. The mixed PROPN/amod solution seems internally inconsistent to me.

Personally I would be happy to call them ADJs and use amod, but I realize the legacy factor here could make it difficult to implement in EWT.

**Comment by nschneid** [comment]
>   * "For organization names with clear syntactic modification structure, the dependencies should also reflect the syntactic modification structure using regular syntactic relations"
>       https://universaldependencies.org/u/dep/flat.html
> 
> 
> The example given in the guidelines is "Natural/PROPN/amod Resources Conservation Service", which I take to be analogous to the FBI example.

Sure but the guidelines don't actually specify PROPN. I take it the ideal pure-UD solution would be to tag "Natural" as ADJ.

**Comment by amir-zeldes** [comment]
Notwithstanding conj and some other wrinkles, it would probably be fairly easy to change upos PROPN to ADJ wherever deprel is amod, but I would not want to change xpos to JJ due to the legacy issues you mentioned and compatibility with large non-UD benchmark datasets (PTB, OntoNotes etc.)

At the same time it seems totally wrong to me to say that "Federal Bureau" is a noun-noun compound!

**Comment by amir-zeldes** [comment]
> The mixed PROPN/amod solution seems internally inconsistent to me.

This is a broader issue IMO, since being a name is not exactly a morphosyntactic category (though it does interact with determiners and some other constructions in many languages). It depends whether you understand PROPN like PTB (=this is a non-function word part of a name) or as a morphological subset of nouns (those things which are both nouns and part of a name). 

If you take the latter stance, then many other changes would need to happen beyond amod/ADJ, including all of the verbs and other content words tagged as PROPN in accordance with the PTB NNP guidelines.

**Comment by nschneid** [comment]
If it were just GUM, I think PROPN->ADJ would be the right solution (keeping NNP for PTB compatibility). What about EWT though? Not sure if anyone will have the appetite to take that on.

**Comment by nschneid** [comment]
The [`PROPN` guidelines](https://universaldependencies.org/u/pos/PROPN.html):

> Note that `PROPN` is only used for the subclass of nouns that are used
as names and that often exhibit special syntactic properties (such as
occurring without an article in the singular in English). When other
phrases or sentences are used as names, the component words retain
their original tags. For example, in _Cat on a Hot Tin Roof_, _Cat_ is
[NOUN](), _on_ is [ADP](), _a_ is [DET](), etc.
> 
> A fine point is that it is not uncommon to regard words that are
etymologically adjectives or participles as proper nouns when they
appear as part of a multiword name that overall functions like a
proper noun, for example in _the Yellow Pages_, _United Airlines_ or
_Thrall Manufacturing Company_.  This is certainly the practice for
the English Penn Treebank tag set. However, the practice should not
be copied from English to other languages if it is not linguistically
justified there. For example, in Czech, _Spojené státy_ “United States”
is an adjective followed by a common noun; their tags in UD are [ADJ]() [NOUN]()
and the adjective modifies the noun via the [amod]() relation.

If I am reading this correctly the second paragraph carves out an exception for **adjectives or participles**, to allow flexibility for New/PROPN York, but it does not express any such exception for "Unite" in "Unite the Union".

When the first paragraph speaks of nouns with special properties, it is referring just to lexically-denoted names like "John" and "Britain", or also parts of names derived from common nouns, like "Republic" in "Central African Republic"? I think most people would assume the latter interpretation. But then we're talking about a phrasal status of entity-ness, rather than a word-level property, which is the crux of the problem.

I guess my gut feeling is that PROPN is kind of problematic as a POS and so we should be conservative about using it and limit it to things that are clearly nouns. To me that seems more intuitive than having all lexical words in a name pretend to be nouns at the POS level while behaving as verbs or adjectives in the dependencies.

But I'm open to hearing other ideas that would provide a clear rationale for various decisions.

**Comment by nschneid** [comment]
P.S. If we were designing this from scratch I would suggest putting phrasal entity information as dependency subtypes, i.e.

* Central/ADJ/amod:name African/ADJ/amod:name Republic/NOUN
* New/ADJ/amod:name York/PROPN
* Unite/VERB the/DET/det Union/NOUN/obj:name

But this would be rather a large change to implement at this point.

**Comment by amir-zeldes** [comment]
> Not sure if anyone will have the appetite to take that on.

Well, if just changing amod is unfeasible for the English data, then also changing verbs etc in names back into verbs etc on top of that is even more unfeasible, no? How would you do it? 

Plus I think there is something off putting about treating Unite as a verb. Generally we've privileged functions that are closer to the matrix root, like free relatives preferring their external function as a label. Doesn't the same logic suggest that Unite is nsubj above?

**Comment by nschneid** [comment]
Good point, if we could call it "subj" it would be fine but it's kind of weird to call it nsubj+VERB or csubj. I feel like UD has painted itself into a corner here.

**Comment by amir-zeldes** [comment]
It depends a little on how literally you understand the label names (don't get me started on advcl). 

I basically understand nsubj to mean nominal (like NP), not necessarily noun. Unite... is a converted VP -> NP so it fits; csubj is a clause, which, at the matrix level, this isn't.

**Comment by nschneid** [comment]
Right, I'm saying that by the lexicalist principles of UD, the deprel encodes nominal vs. clausal AND is expected to match the POS. So a VERB would not be expected to serve as `nsubj`.

For these derived names ideally we want an external/phrasal POS (PROPN) and an internal one (VERB).

You're arguing that the external one should win, which is a clear principle, but then for the internal structure you have a PROPN with an `obj` dependent, which is confusing.

Or I guess the policy could be to fall back on `compound` for the internal structure in such cases...this is giving me flashbacks to UniversalDependencies/docs#753 (phrasal compound modifiers).

**Comment by nschneid** [comment]
@dan-zeman is this mainly a problem for analytic languages like English, or is it more widespread? (Verbal phrases getting repurposed as names without extra marking.)

**Comment by dan-zeman** [comment]
You can have a phrase acting as a nominal even if the phrase is not headed by a `NOUN`/`PRON`/`PROPN`. This is certainly the case with book/movie titles, and it also holds for sentences where a non-nominal word is the topic, e.g., in: _Unite is not an auxiliary verb in English._

If the head of the phrase is a verb, then I think it should be tagged `VERB` but attached as `nsubj`, not `csubj` (provided the phrase is a subject). I think the validator will not complain in the subject case but we probably have a problem anyway, because it does check some other dependent positions. The phrase could be an `obj` (not `ccomp`), or an `obl` (not `advcl`), and I'm afraid the validator will currently report `obl` + `VERB` as an error.

Independently of all the above, one can ask whether _Unite the Union_ is still in the same category as book titles. When googling to find out what the hell the phrase refers to :-) I saw that they often shorten it to just _Unite_ on their web, as in _Show me how Unite helps me_, or _Unite Guide for Officers_. Only a step from a hypothetical sentence like _The local members left Unite to create a new union under the name Unite Ireland, so there are now two <b>Unites</b> operating in Belfast._ Here the morphology would definitely indicate that it is a noun.

Generally, this is something that could happen in any language. I am indeed aware of organizations/movements whose name is a verb (or headed by a verb). In Czech, they retain the verbal morphological form, which is frozen, and the name phrase typically refuses acquiring any nominal morphology. Syntactically it is a noun, which sometimes creates issues with agreement (it does not have gender but we need to know the gender). In such cases it may be preferable to accompany it with a real noun, as in _hnutí Probuďte se!_ "the movement Wake Up!"

**Comment by nschneid** [comment]
What about adding a feature for such cases, e.g. `CoercedNominal=Yes`? Then the validator could treat a VERB as nominal with respect to its parent (and a verb with respect to its dependents).

(This could be generalized to allow a phrase to be coerced to any POS: `CoercedPOS=NOUN` for the ones we're discussing, `CoercedPOS=ADJ` for phrasal compound modifiers, `CoercedPOS=ADV` for adverbial NPs, etc., but that might be too radical a change.)

**Comment by dan-zeman** [comment]
`CoercedNominal=Yes` sounds good to me. The generality (and also brevity) of `CoercedPOS=NOUN` is appealing but I don't see a real need for it at present. All we (seem to) need is to be able to say that a word from a non-nominal category is used as a label. Either a label refering to a real-world entity, or as a meta-word, when speaking about the word. The label works as a nominal with respect to its parent, while its morphology and its syntactic relations with its children follow the lexical POS category. If the label is used "adverbially", it can still be a nominal, attached as `obl`.

I don't know whether it would be better to make this a morphological feature or a MISC attribute. As a mere instruction for the validator it would be a fit for MISC. It does say something about the behavior of the word, but it is more about the syntactic context than about this form of the word per se. On the other hand, if it is a language-specific "morphological" feature, people will be forced to document it with examples :-)

One could also say that this is like head promotion in ellipsis (here the elided item would be a common noun that is appositionally modified by the label phrase). The promoted word inherits the deprel of the elided head but it retains its own part of speech. This is the reason why the validator is very limited in checking correlations between UPOS tags and deprels.

Perhaps we should transfer this issue to the main discussion under the docs repository. It is no longer just about a bug in the EWT treebank.
